pkgs/development/libraries/jemalloc/rust.nix

```diff
@@ -0,0 +1,24 @@
+{ lib
+, stdenv
+, jemalloc
+, writeText
+
+, unprefixed ? false
+}:
+
+let
+  # On some platforms the unprefixed feature will be ignored:
+  # https://github.com/tikv/jemallocator/blob/ab0676d77e81268cd09b059260c75b38dbef2d51/jemalloc-sys/src/env.rs
+  unprefixed' = unprefixed && !stdenv.hostPlatform.isMusl && !stdenv.hostPlatform.isDarwin && !stdenv.hostPlatform.isAndroid;
+
+in jemalloc.overrideAttrs (oldAttrs: {
+  configureFlags = oldAttrs.configureFlags ++ lib.optionals (!unprefixed') [
+    "--with-jemalloc-prefix=_rjem_"
+  ] ++ [
+    "--with-private-namespace=_rjem_"
+  ];
+
+  setupHook = writeText "setup-hook.sh" ''
+    export JEMALLOC_OVERRIDE="@out@/lib/libjemalloc${stdenv.hostPlatform.extensions.library}"
+  '';
```

Comment:
I wonder, what if we put this in the normal jemalloc package?  It seems a shame to rebuild (for unprefixed) _just_ to add a setup hook.

Reply:
The unprefixed variant still needs a `"--with-private-namespace=_rjem_"`

Reply:
Ahh I misread.  Maybe you could swap the order --with-jemalloc-prefix and --with-private-namespace, so it's clearer that only --with-jemalloc-prefix is conditionally applied?

Reply:
done